AfrikaBurn/MapViewController.swift

```diff
@@ -0,0 +1,194 @@
+//
+//  MapViewController.swift
+//  AfrikaBurn
+//
+//  Created by Daniel Galasko on 2017/03/04.
+//  Copyright © 2017 AfrikaBurn. All rights reserved.
+//
+
+import UIKit
+import MapKit
+import RealmSwift
+
+struct Resources {
+    static let mapOverlayImage: UIImage = {
+        let path = Bundle.main.path(forResource: "AB2017_Layout_RoadsLabel_20170308", ofType: "png")!
+        return UIImage(contentsOfFile: path)!
+    }()
+}
+
+struct BurnMap {
+    static let boundingMapRect: MKMapRect = MKMapRect(origin: MKMapPoint(x: 148937508.42330855, y: 159711548.54660633), size: MKMapSize(width: 9388.6817751824855, height: 7286.825931340456))
+}
+
+class MapViewController: UIViewController {
+    
+    var selectedElement : AfrikaBurnElement? = nil
+    @IBOutlet weak var mapView: MKMapView!
+    let persistentStore = PersistentStore()
+    let locationManager = CLLocationManager()
+    lazy var allElements: Results<AfrikaBurnElement> = self.persistentStore.elements()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        locationManager.requestWhenInUseAuthorization()
+        mapView.showsUserLocation = true
+        mapView.delegate = self
+    
+        loadElements()
+        
+        self.navigationItem.title = "Map"
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        loadElements()
+    }
+    
+    func loadElements(){
+        
+        for element in allElements {
+            guard let location = element.location else {
+                continue
+            }
+            let anno = BurnAnnotation(coordinate: location);
+            anno.element = element
+            anno.title = element.name
+            anno.image = UIImage(named: "map-pin.png")
```

Comment:
Since this image is in the asset catalog you can actually reference it as `map-pin` without the suffix. Taking that a step further you can also use the image literal syntax and just type `map-pin` without the `UIImage` constructor. Food for thought 😄